device/timers.go

```diff
@@ -100,11 +102,11 @@ func expiredRetransmitHandshake(peer *Peer) {
 		peer.device.log.Verbosef("%s - Handshake did not complete after %d seconds, retrying (try %d)", peer, int(RekeyTimeout.Seconds()), peer.timers.handshakeAttempts.Load()+1)
 
 		/* We clear the endpoint address src address, in case this is the cause of trouble. */
-		peer.Lock()
-		if peer.endpoint != nil {
-			peer.endpoint.ClearSrc()
+		ep, _ := peer.endpoint.Load().(conn.Endpoint)
+		if ep != nil {
+			ep.ClearSrc()
+			peer.endpoint.Store(ep)
```

Comment:
This line does nothing? You just loaded it from peer.endpoint so why are you storing it back?